sdk/search/search-documents/src/searchIndexingBufferedSender.ts

```diff
@@ -218,6 +326,9 @@ export class SearchIndexingBufferedSender<T> {
     retryAttempt: number = 0
   ): Promise<void> {
     try {
+      for (let i = 0; i < actionsToSend.length; i++) {
+        this.emitter.emit("batchSent", actionsToSend[i]);
```

Comment:
`batchSent` fires on every single document? Shouldn't it be `documentSent` then?

Also, it feels weird to fire this *before* the document is actually sent. Should it be called `beforeDocumentSent` or something instead to indicate that?

Reply:
Agreed. But it was a specific ask that I need to send it for each document. I have changed the name to ```beforeDocumentSent``` in the code and the samples